src/wallet/wallet.cpp

```diff
@@ -2042,6 +2048,9 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int
         if (output.nDepth < (pcoin->IsFromMe(ISMINE_ALL) ? nConfMine : nConfTheirs))
             continue;
 
+        if (pcoin->GetMempoolAncestorCount() > nMaxAncestors)
```

Comment:
I think you need the pcoin->InMempool() check you had before, in case you're looking at in-chain coins?